### DISCUSSION TITLE
Any interest in an Arduino CLI Github Action?

### DISCUSSION THREAD

**Discussion by witnessmenow** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#457] [Ideas]
I recently added this to one of my own libraries and it seems to work well

https://github.com/witnessmenow/file-fetcher-arduino/blob/main/.github/workflows/arduino_build.yml

I don't mind adding a PR for it if there is any demand.

Not sure if there is any advantage over the current PIO build though other than this library doesn't actually seem to be compiling with the arduino CLI as it has some errors and warnings

```
/home/runner/Arduino/libraries/ESP32_HUB75_LED_MATRIX_PANEL_DMA_Display/src/ESP32-HUB75-MatrixPanel-I2S-DMA.cpp:195:7: error: variable 'num_dma_payload_colour_depths' set but not used [-Werror=unused-but-set-variable]
     int num_dma_payload_colour_depths = m_cfg.getPixelColorDepthBits();
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cc1plus: some warnings being treated as errors
```

Which does seem to be a valid warning

**Comment by mrcodetastic** [comment]
Hi Brian,

This would indeed be useful I think. I've been caught out making changes to this library that compiled with pio but not Arduino due to "strictness"

Thanks.

**Reply by witnessmenow** [reply]
Cool, is there a formatter you use for code? The one I use has made lots of changes, but I can just reply my changes on a fresh copy without the formatting changes

E.g: https://github.com/witnessmenow/ESP32-HUB75-MatrixPanel-I2S-DMA/commit/45f1e46a7470d82e47356ef5d866d9ebf3bfa20d

**Reply by witnessmenow** [reply]
I have it working for all the examples other than the two that use the animated Gif library, error comes from that library though.

**Reply by mrcodetastic** [reply]
Thanks for this. Can you restrict the CI to only run on the "simple test shapes" example? That will be enough, and it minimises the time it takes for the CI to run after a commit has been made.

**Reply by witnessmenow** [reply]
Yeah for sure, I'll do it over the weekend

**Reply by mrcodetastic** [reply]
No problems.
